src/search-manager/SearchManager.ts

```diff
@@ -36,19 +39,87 @@ export interface SearchContextData {
+  // Returns the string equivalent of the given Type value.
+  // Returning empty string/undefined will clear the url search paramName.
+  toString: (value: Type) => string | undefined,
+): [value: Type, setter: React.Dispatch<React.SetStateAction<Type>>] {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [stateValue, stateSetter] = React.useState<Type>(defaultValue);
+
+  // The converted search parameter value takes precedence over the state value.
+  const returnValue: Type = fromString(searchParams.get(paramName)) ?? stateValue ?? defaultValue;
+
+  // Before updating the state value, update the url search parameter
+  const returnSetter: React.Dispatch<React.SetStateAction<Type>> = ((value: Type) => {
+    const paramValue: string = toString(value) ?? '';
+    if (paramValue) {
+      searchParams.set(paramName, paramValue);
+    } else {
+      // If no paramValue, remove it from the search params, so
+      // we don't get dangling parameter values like ?paramName=
+      // Another way to decide this would be to check value === defaultValue,
+      // and ensure that default values are never stored in the search string.
+      searchParams.delete(paramName);
+    }
+    setSearchParams(searchParams, { replace: true });
+    return stateSetter(value);
+  });
+
+  // Return the computed value and wrapped set state function
+  return [returnValue, returnSetter];
+}
+
 export const SearchContextProvider: React.FC<{
   extraFilter?: Filter;
+  overrideSearchSortOrder?: SearchSortOption
   children: React.ReactNode,
   closeSearchModal?: () => void,
-}> = ({ extraFilter, ...props }) => {
+}> = ({ overrideSearchSortOrder, ...props }) => {
   const [searchKeywords, setSearchKeywords] = React.useState('');
   const [blockTypesFilter, setBlockTypesFilter] = React.useState<string[]>([]);
   const [tagsFilter, setTagsFilter] = React.useState<string[]>([]);
+  const extraFilter: string[] = forceArray(props.extraFilter);
+
+  // The search sort order can be set via the query string
+  // E.g. ?sort=display_name:desc maps to SearchSortOption.TITLE_ZA.
+  const defaultSortOption = SearchSortOption.RELEVANCE;
+  const [searchSortOrder, setSearchSortOrder] = useStateWithUrlSearchParam<SearchSortOption>(
+    defaultSortOption,
+    'sort',
+    (value: string) => Object.values(SearchSortOption).find((enumValue) => value === enumValue),
+    (value: SearchSortOption) => value.toString(),
+  );
+  // SearchSortOption.RELEVANCE is special, it means "no custom sorting", so we
+  // send it to useContentSearchResults as an empty array.
+  const searchSortOrderToUse = overrideSearchSortOrder ?? searchSortOrder;
+  const sort: SearchSortOption[] = (searchSortOrderToUse === defaultSortOption ? [] : [searchSortOrderToUse]);
+  // Selecting SearchSortOption.RECENTLY_PUBLISHED also excludes unpublished components.
+  if (searchSortOrderToUse === SearchSortOption.RECENTLY_PUBLISHED) {
+    extraFilter.push('last_published IS NOT EMPTY');
```

Comment:
Can you try this @pomegranited?
```suggestion
    extraFilter.push('last_published IS NOT NULL');
```

Reply:
🤦‍♀️ That was totally it.. Thank you @rpenido !

Reply:
And re the `last_published` date stored in the index, I think I've found the problem, but am not sure how to address it, so have [asked Dave](https://github.com/openedx/edx-platform/pull/35103/files#r1693274718).